src/huggingface_hub/fastai_utils.py

```diff
@@ -0,0 +1,393 @@
+            Directory where model card is created.
+    """
+    readme_path = repo_dir / "README.md"
+
+    if not readme_path.exists():
+        with readme_path.open("w", encoding="utf-8") as f:
+            f.write(README_TEMPLATE)
+
+
+def _create_model_pyproject(repo_dir: Path):
+    """
+    Creates a `pyproject.toml` for the repository.
+
+    Params:
+        repo_dir (:obj:`Path`):
+            Directory where `pyproject.toml` is created.
+    """
+    pyproject_path = repo_dir / "pyproject.toml"
+
+    if not pyproject_path.exists():
+        with pyproject_path.open("w", encoding="utf-8") as f:
+            f.write(PYPROJECT_TEMPLATE)
+
+
+def save_fastai_learner(
+    learner,
+    save_directory: str,
+    config: Optional[Dict[str, Any]] = None,
+):
+    """
+    Saves a fastai learner to `save_directory` in pickle format using the default pickle protocol for the version of python used.
+    Apply this if you're using Learners.
+
+    Params:
+        learner (:obj:`Learner`):
+            The `fastai.Learner` you'd like to save.
+        save_directory (:obj:`str`):
+            Specific directory in which you want to save the fastai learner.
+        config (:obj:`dict`, `optional`):
+            Configuration object. Will be uploaded as a .json file. Example: 'https://huggingface.co/espejelomar/fastai-pet-breeds-classification/blob/main/config.json'.
+
+    Raises:
+        :class:`RuntimeError`: if the config file provided is not a dictionary.
+    """
+    # Check that fastai and fastcore versions are supported.
+    check_fastai_fastcore_versions()
+
+    os.makedirs(save_directory, exist_ok=True)
+
+    # save config
+    # if user provides config then we update it with the fastai and fastcore versions in CONFIG_TEMPLATE.
+    if config is not None:
+        if not isinstance(config, dict):
+            raise RuntimeError(
+                f"Provided config should be a dict. Got: '{type(config)}'"
+            )
+        path = os.path.join(save_directory, CONFIG_NAME)
+        with open(path, "w") as f:
+            json.dump(config, f)
+
+    # creating README.md if none exist
+    _create_model_card(Path(save_directory))
+
+    # creating pyproject.toml if none exist
+    _create_model_pyproject(Path(save_directory))
+
+    # saving learner
+    # learner.export saves the model in `self.path/save_directory` and this folder should exist.
+    # We create this folder and call it `fastai_path`
+    fastai_path = os.path.join(learner.path, save_directory)
+    os.makedirs(fastai_path, exist_ok=True)
+    learner.export(
+        fname=os.path.join(save_directory, "model.pkl"),
+        pickle_protocol=DEFAULT_PROTOCOL,
+    )
+
+    # We move the model from `self.path/save_directory/model.pkl` to `save_directory/model.pkl`.
+    os.rename(
+        os.path.join(fastai_path, "model.pkl"),
+        os.path.join(save_directory, "model.pkl"),
+    )
+
+
+def from_pretrained_fastai(
+    model_id: str,
```

Comment:
Does this not work on local files? I believe the keras + pytorch integrations both do, which is why I suggest this one should too.